src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/service/KeyService.java

```diff
@@ -137,5 +155,52 @@ static boolean isCausedByKeyNotFound(CodedException e) {
         return KEY_NOT_FOUND_FAULT_CODE.equals(e.getFaultCode());
     }
 
-    static final String KEY_NOT_FOUND_FAULT_CODE = SIGNER_X + "." + X_KEY_NOT_FOUND;
+    static boolean isCausedByCsrNotFound(CodedException e) {
+        return CSR_NOT_FOUND_FAULT_CODE.equals(e.getFaultCode());
+    }
+
+    private static String signerFaultCode(String detail) {
+        return SIGNER_X + "." + detail;
+    }
+
+    static final String KEY_NOT_FOUND_FAULT_CODE = signerFaultCode(X_KEY_NOT_FOUND);
+    static final String CSR_NOT_FOUND_FAULT_CODE = signerFaultCode(X_CSR_NOT_FOUND);
+
+    public void deleteCsr(String keyId, String csrId) throws KeyNotFoundException, CsrNotFoundException {
+        KeyInfo keyInfo = getKey(keyId);
+        CertRequestInfo csrInfo = getCsr(keyInfo, csrId);
```

Comment:
Is this line necessary? crsInfo is not used at all.

Reply:
To me it seems it is necessary for throwing exception if csr is not found. I guess declaring a variable could be omitted.

Reply:
Yes, that was the reason. Removed variable and added a comment.